msu/utilities/misc.nut

```diff
@@ -57,3 +57,102 @@
 
 	return ::isKindOf(obj, _className);
 }
+
+::MSU.serializeObject <- function( _object, _out )
+{
+	::MSU.requireOneOfType(["array", "table"], _object);
+	
+	local type = typeof _object;
+	local isTable = type == "table";
+
+	_out.writeString(type);
+	_out.writeU32(_object.len());
+
+	foreach (key, element in _object)
+	{
+		if (isTable) _out.writeString(key);
+		local elementType = typeof element;
+		_out.writeString(elementType);
+
+		switch (elementType)
+		{
+			case "integer":
+				if (element < 0)
+				{
+					_out.writeString("I");
+					_out.writeI32(element);
+				}
+				else
+				{
+					_out.writeString("U");
+					_out.writeU32(element);
+				}
+				break;
```

Comment:
I think it'd be cool if we could use a boolean here since this writes to raw bytes and that actually makes a difference in the size of the final file.

Reply:
Or at least a very short int if need be

Reply:
Will go with bool